app/src/main/java/pl/zgora/uz/pum/QRCodeActivity.java

```diff
@@ -4,42 +4,50 @@
 import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.support.v7.widget.CardView;
 import android.view.View;
 import android.widget.Button;
+import android.widget.GridLayout;
 import android.widget.Toast;
 
 import com.google.zxing.integration.android.IntentIntegrator;
 import com.google.zxing.integration.android.IntentResult;
 
 public class QRCodeActivity extends AppCompatActivity {
-    private Button scanButton, dataActivityButton;
+
+    private GridLayout mainGrid;
+    final Activity activity = this;
```

Comment:
(Optional) Field might be set to `private static final Activity ACTIVITY = this;`